internal/util/GitCliUtil.go

```diff
@@ -0,0 +1,105 @@
+package util
+
+import (
+	"fmt"
+	"go.uber.org/zap"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/config"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+type GitCliUtil struct {
+	logger *zap.SugaredLogger
+}
+
+func NewGitCliUtil(logger *zap.SugaredLogger) *GitCliUtil {
+	return &GitCliUtil{
+		logger: logger,
+	}
+}
+
+const GIT_ASK_PASS = "/git-ask-pass.sh"
+
+func (impl *GitCliUtil) Fetch(rootDir string, username string, password string) (response, errMsg string, err error) {
+	impl.logger.Debugw("git fetch ", "location", rootDir)
+	cmd := exec.Command("git", "-C", rootDir, "fetch", "origin", "--tags", "--force")
+	output, errMsg, err := impl.runCommandWithCred(cmd, username, password)
+	impl.logger.Debugw("fetch output", "root", rootDir, "opt", output, "errMsg", errMsg, "error", err)
+	return output, errMsg, nil
```

Comment:
we are not returning err, is it intentional? same for other functions too

Reply:
we are checking if err != nil and then if errMsg != nil, we should do write that code here and change signature to return (response string, error)

Reply:
same for other functions too